math/src/field/fields/u384_prime_field.rs

```diff
@@ -0,0 +1,208 @@
+use std::fmt::Debug;
+use std::marker::PhantomData;
+
+use crate::field::element::FieldElement;
+use crate::field::traits::IsField;
+use crate::unsigned_integer::UnsignedInteger384 as U384;
+
+pub trait HasU384Constant: Debug + Clone + Eq + PartialEq {
+    const VALUE: U384;
+}
```

Comment:
Let's add a comment explaining that this trait is necessary while there is no way in Rust of using a non-native const value as a generic so it could be revisited later (this will be possible in the near future).

Also, we will need a similar trait for `U256` so it would be better if we generalize this trait for the type of the order. It could be something like this: 
```suggestion
/// This trait is necessary for us to be able to use unsigned integer types bigger than 
/// `u128` (the biggest native `unit`) as constant generics.
/// This trait should be removed when Rust supports this feature.
pub trait BigOrder<T>: Debug + Clone + Eq + PartialEq {
    const VALUE: T;
}
```

This way we could use this trait for any non-native type as a constant generic for the order.

Reply:
Thanks. Added!